src/main/java/frc/robot/commands/DefaultDriveCommand.java

```diff
@@ -39,15 +42,12 @@ public void execute() {
         rotation = rotationSupplier.getAsDouble();
         
         drivetrainSubsystem.drive(x, y, rotation);
-        
-        if (rotation == 0 && (x != 0 || y != 0)) {
-            if (updateRotationTarget) {
-                drivetrainSubsystem.setRotationTarget(rotationTarget);
-                updateRotationTarget = false;
-            }
-        } else {
+
+        if (rotation != 0) {
+            target = drivetrainSubsystem.getRotationTarget();
             drivetrainSubsystem.setRotationTarget(null);
-            updateRotationTarget = true;
+        } else {
+            drivetrainSubsystem.setRotationTarget(target);
```

Comment:
Overall changes look good. The restoring of the target back to when not rotating is an interesting idea but I feel like there might be times when we don't want to be oriented only 0 or 180. 

So my original comment was trying to suggest that any movement of the rotation joystick just cancels the targetRotation entirely. Driver would have to press the button again to reset it.

The use case I have in mind is if we are doing ground pickup of an object and need to approach it from a unique angle. Also having the extra spin after the driver lets go of the joystick could have some weird consequences (might rotate suddenly back an opposite direction).

Reply:
Sorry, I think I misunderstood your original feedback